roles/add-nexus-repo/tests/test.yml

```diff
@@ -0,0 +1,162 @@
+---
+# This test assumes a locally running Nexus3 app.
+# The easiest way to acheive that is via `docker run -it -p 8081:8081 sonatype/nexus3
+
+- name: 'Add Nexus Repos'
+  hosts: ocp-nexus
+  vars:
+    nexus_api_base_path: '/service/rest/v1'
+    nexus_user: admin
+    nexus_password: 'admin123'
+    nexus_protocol: http
+    nexus_url: 'localhost:8081'
+    nexus_server_poll_retries: 100
+    nexus_server_poll_delay_in_seconds: 10
+  tasks:
+  - name: Start nexus container
+    command: 'docker run -d --name ansible_nexus_test -p 8081:8081 sonatype/nexus3'
+  - name: "Ensure the Nexus Server is Up"
+    uri:
+      url: "{{ nexus_protocol }}://{{ nexus_url }}/service/metrics/healthcheck"
```

Comment:
Found something interesting while reviewing this. So when nexus initially stands up and you try to use `service/metrics/healthcheck` as the url to check, this will fail. You get a response like you would expect, but instead of a `200`, you get a `500` because the default admin credentials haven't been changed at this point (which is probably the wrong http code to throw, but that's a different topic).

So I'm thinking rather than using the `healthcheck` endpoint and having to add some additional logic, we can probably just rely on `service/metrics/ping` as it seems to provide the functionality that we're looking for (i.e. app is up and healthy)

Reply:
Yeah, I tried this as well and got the same result. `service/metrics/ping` seems to resolve this.

Reply:
Unfortunately, this seems to be a moving target for a little while. I am working on a new role which will automatically detect the Nexus version and use the appropriate URLs and settings. Looks like big, and awesome, changes are coming in the next few Nexus releases which should make a lot of this simpler.